app/forms/system_admin/change_accredited_provider_form.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module SystemAdmin
+  class ChangeAccreditedProviderForm < TraineeForm
+    FIELDS = %i[
+      accredited_provider_id
+      audit_comment
+      zendesk_ticket_url
+    ].freeze
+
+    attr_accessor(*FIELDS)
+    attr_accessor :step
+
+    validates :accredited_provider_id, presence: true, if: -> { !step || step == :provider }
+    validates :audit_comment, presence: true, if: -> { !step || step == :reasons }
```

Comment:
The idea here is that the two validations are applied only on the relevant input form and then once more on the confirmation page.